# Atividade em Grupo (TREINO) — Análise de Mercado de Combustíveis (SP, 2025)

Versão de treino gerada a partir do gabarito: **soluções removidas**, enunciados e saídas esperadas mantidos.

**Requisito:** `postos_sp_2025.csv` e `distribuicao_municipios_sp_2025.csv` na MESMA pasta deste notebook.

Os blocos "✅ Confira" são as saídas do professor sobre os dados reais: rode seu código e compare. Gabarito completo continua em `AGP2025_GABARITO.ipynb`, só abrir depois de tentar.

---

In [1]:
import pandas as pd

---
## Questão 1 — Conhecendo os Dados _(2,0 pontos)_

### 1a) _(0,5 pt)_ — Carregar e descrever o dataset

In [26]:
# ESPAÇO PARA RESPOSTA DA QUESTAO 1a - NÃO DELETAR ESTA LINHA !!

postos = pd.read_csv("postos_sp_2025.csv", sep= ";",
decimal= ",")
postos.columns
postos.head()

,Regiao,Estado,Municipio,Revenda,CNPJ,Produto,Data da Coleta,Valor de Venda,Bandeira
0,SE,SP,SAO PAULO,AUTO POSTO MISTRAL LTDA.,04.937.358/0001-04,GASOLINA,01/01/2025,5.99,VIBRA
1,SE,SP,SAO PAULO,AUTO POSTO MISTRAL LTDA.,04.937.358/0001-04,ETANOL,01/01/2025,3.99,VIBRA
2,SE,SP,SAO PAULO,AUTO POSTO DUQUE CITY LTDA,04.765.173/0001-51,GASOLINA,01/01/2025,7.99,VIBRA
3,SE,SP,SAO PAULO,AUTO POSTO DUQUE CITY LTDA,04.765.173/0001-51,ETANOL,01/01/2025,5.89,VIBRA
4,SE,SP,SAO PAULO,AUTO POSTO NOVO CENTRO LTDA,02.646.144/0001-45,GASOLINA,01/01/2025,5.79,VIBRA


✅ Confira: `(76934, 9)` — colunas: `Regiao, Estado, Municipio, Revenda, CNPJ, Produto, Data da Coleta, Valor de Venda, Bandeira`.

### 1b) _(0,5 pt)_ — Registros por produto

In [27]:
# ESPAÇO PARA RESPOSTA DA QUESTAO 1b - NÃO DELETAR ESTA LINHA !!

postos["Produto"].value_counts()

Produto
GASOLINA      29072
ETANOL        28770
DIESEL S10    19092
Name: count, dtype: int64

✅ Confira: `GASOLINA 29072`, `ETANOL 28770`, `DIESEL S10 19092`.

### 1c) _(1,0 pt)_ — Mín, média e máx da Gasolina em uma operação

In [42]:
postos_gasolina = postos[postos["Produto"] == "GASOLINA"]
postos_gasolina["Valor de Venda"].agg(["min", "mean", "max"])

min     4.990000
mean    6.122167
max     8.990000
Name: Valor de Venda, dtype: float64

✅ Confira: `min 4.99`, `mean 6.12`, `max 8.99`.

---
## Questão 2 — Álcool ou Gasolina? _(4,0 pontos)_

### 2a) _(0,5 pt)_ — Criar a coluna `Mes` (YYYY-MM)

In [45]:
# ESPAÇO PARA RESPOSTA DA QUESTAO 2a - NÃO DELETAR ESTA LINHA !!

postos.head()
postos["Mes"] = postos["Data da Coleta"].str[6:10] + "-" + postos["Data da Coleta"].str[3:5]
postos.head()

,Regiao,Estado,Municipio,Revenda,CNPJ,Produto,Data da Coleta,Valor de Venda,Bandeira,Mes
0,SE,SP,SAO PAULO,AUTO POSTO MISTRAL LTDA.,04.937.358/0001-04,GASOLINA,01/01/2025,5.99,VIBRA,2025-01
1,SE,SP,SAO PAULO,AUTO POSTO MISTRAL LTDA.,04.937.358/0001-04,ETANOL,01/01/2025,3.99,VIBRA,2025-01
2,SE,SP,SAO PAULO,AUTO POSTO DUQUE CITY LTDA,04.765.173/0001-51,GASOLINA,01/01/2025,7.99,VIBRA,2025-01
3,SE,SP,SAO PAULO,AUTO POSTO DUQUE CITY LTDA,04.765.173/0001-51,ETANOL,01/01/2025,5.89,VIBRA,2025-01
4,SE,SP,SAO PAULO,AUTO POSTO NOVO CENTRO LTDA,02.646.144/0001-45,GASOLINA,01/01/2025,5.79,VIBRA,2025-01


✅ Confira: `01/01/2025 -> 2025-01` nas 3 primeiras linhas.

### 2b) _(1,5 pt)_ — Preço médio por Município, Mês e Produto (`comparacao`)

In [55]:
# ESPAÇO PARA RESPOSTA DA QUESTAO 2b - NÃO DELETAR ESTA LINHA !!

ge = postos[postos["Produto"].isin(["GASOLINA", "ETANOL"])]

comparacao = ge.pivot_table(index=["Municipio", "Mes"],   # Linhas (lista: 2 níveis)
                            columns="Produto",            # Colunas
                            values="Valor de Venda",      # Valores
                            aggfunc="mean").reset_index() # média + vira coluna comum
comparacao.columns.name = None                            # limpa o rótulo do cabeçalho
comparacao.head() 

,Municipio,Mes,ETANOL,GASOLINA
0,ADAMANTINA,2025-01,3.896250,5.935000
1,ADAMANTINA,2025-02,3.984688,6.009375
2,ADAMANTINA,2025-03,3.987500,6.078125
3,ADAMANTINA,2025-04,3.942250,6.027000
4,ADAMANTINA,2025-05,3.914516,6.051875


✅ Confira: DataFrame com `522 linhas × 4 colunas` (`Municipio, Mes, ETANOL, GASOLINA`).
Primeira linha: `ADAMANTINA | 2025-01 | 3.90 | 5.94`.

### 2c) _(1,0 pt)_ — Razão e combustível vantajoso

In [ ]:
# ESPAÇO PARA RESPOSTA DA QUESTAO 2c - NÃO DELETAR ESTA LINHA !!



✅ Confira (5 primeiras linhas):** todas com `Razao ≈ 0.65` e `Combustivel_Vantajoso = "Etanol"`.
No total: **514** combinações favorecem o Etanol e **8** a Gasolina.

### 2d) _(1,0 pt)_ — Municípios com maior margem para aumento de preço

In [ ]:
# ESPAÇO PARA RESPOSTA DA QUESTAO 2d - NÃO DELETAR ESTA LINHA !!



✅ Confira (menor → maior razão média):**

| Município | Razão média |
|---|---|
| SAO JOSE DO RIO PRETO | 0.6259 |
| ARACATUBA | 0.6279 |
| BARUERI | 0.6320 |
| BIRIGUI | 0.6347 |
| VOTUPORANGA | 0.6391 |

---
## Questão 3 — Margem Comercial dos Postos _(4,0 pontos)_

### 3a) _(0,5 pt)_ — Carregar distribuição e fazer o merge (`dados`)

In [ ]:
# ESPAÇO PARA RESPOSTA DA QUESTAO 3a - NÃO DELETAR ESTA LINHA !!



✅ Confira: `(76934, 11)` — nenhuma linha é perdida no merge (todas as combinações encontram correspondência na distribuição).

### 3b) _(1,0 pt)_ — Coluna `Margem` e estatísticas globais

In [ ]:
# ESPAÇO PARA RESPOSTA DA QUESTAO 3b - NÃO DELETAR ESTA LINHA !!



✅ Confira: `min -0.28`, `mean 0.74`, `max 3.88`.

### 3c) _(0,5 pt)_ — Margem média por bandeira (Gasolina)

In [ ]:
# ESPAÇO PARA RESPOSTA DA QUESTAO 3c - NÃO DELETAR ESTA LINHA !!



✅ Confira (maior → menor, R$/L):** `TAURUS 1.06`, `VIBRA 0.98`, `IPIRANGA 0.85`, `RAIZEN 0.84`, … , `BRANCA 0.54`, `RODOIL 0.26`.

### 3d) _(1,5 pt)_ — Decisão de investimento: VIBRA vs BRANCA

In [ ]:
# ESPAÇO PARA RESPOSTA DA QUESTAO 3d - NÃO DELETAR ESTA LINHA !!



✅ Confira: diferença por litro ≈ **R$ 0,4435**; diferença mensal ≈ **R$ 22.175,43**.

In [ ]:
# ESPAÇO PARA RESPOSTA DA QUESTAO 3d - NÃO DELETAR ESTA LINHA !!

